src/routers/posts.ts

```diff
@@ -22,25 +22,1077 @@ import {
+ *       - in: query
+ *         name: categories
+ *         schema:
+ *           type: string
+ *         description: Comma-separated list of categories to filter posts by.
+ *     responses:
+ *       200:
+ *         description: A list of posts matching the search criteria.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 count:
+ *                   type: integer
+ *                   description: The number of posts returned.
+ *                 posts:
+ *                   type: array
+ *                   items:
+ *                     type: object
+ *                     properties:
+ *                       _id:
+ *                         type: string
+ *                         description: The unique identifier of the post.
+ *                       title:
+ *                         type: string
+ *                         description: The title of the post.
+ *                       price:
+ *                         type: number
+ *                         description: The price of the post.
+ *                       category_url:
+ *                         type: string
+ *                         description: The URL slug of the category.
+ *                       createdAt:
+ *                         type: string
+ *                         format: date-time
+ *                         description: The creation timestamp of the post.
+ *                       creator_id:
+ *                         type: object
+ *                         properties:
+ *                           _id:
+ *                             type: string
+ *                             description: The unique identifier of the creator.
+ *                           username:
+ *                             type: string
+ *                             description: The username of the post creator.
+ *                       likes:
+ *                         type: array
+ *                         items:
+ *                           type: object
+ *                           properties:
+ *                             _id:
+ *                               type: string
+ *                               description: The unique identifier of the like.
+ *                             user_id:
+ *                               type: string
+ *                               description: The ID of the user who liked the post.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 err:
+ *                   type: string
+ *                   description: Error message
+ */
 router.get('/search', searchPosts);
+
+/**
+ * @swagger
+ * /posts/checkLikes/{postID}:
+ *   get:
+ *     tags:
+ *       - Posts operations
+ *     summary: Count the number of likes for a specific post
```

Comment:
summary should contains just -
"Must be connected as admin" OR
"Must be connected as user"